### PR TITLE
fix:issue on do not validate locale if not creating for all locales

### PR DESCRIPTION
**Pull request by binar1**
### What does it do?
After applying a change this [pull request ](https://github.com/strapi/strapi/pull/19626) some cases has been missed like for example when the local is not default one and you want to create and publish 

### Why is it needed?
Because the user can not create or save any single type  module so i just add another validation to check if the local exist or not

### How to test it?
Create single types and collection type with localization enabled , look to you database you can see all data will be create depend of your current local and when you are going to publish it it says 404 because it can not find the record with that local in the database

### Related issue(s)/PR(s)

- [#19798](https://github.com/strapi/strapi/issues/19798)
- [#19715](https://github.com/strapi/strapi/issues/19715)
-  [#19626](https://github.com/strapi/strapi/pull/19626) 
- [#19805](https://github.com/strapi/strapi/issues/19805)

### CODE REVIEW

**Review by Marc-Roig — DISMISSED** [review]
thank you for the fix 🚀

**Comment by Marc-Roig** [comment]
@binar1  could you run prettier to format the file you updated? 🙏 (yarn prettier:code)

**Comment by binar1** [comment]
Hello @Marc-Roig  noted. I've pushed a new commit where Prettier has been used.

**Comment by Marc-Roig** [comment]
thank you! will run the CI and once its green will merge this work, did a quick check and seems to be working, but we will run it through a QA process during the week and release this on `4.21.1` . Thank you again!

File: packages/plugins/i18n/server/src/register.ts [hunk 1]
```diff
@@ -30,8 +30,17 @@ const isUrlForCreation = (url: string) => {
...
   const splitUrl = url.split('/').filter(Boolean);
 
+  // Checking for query params and removed in the
+  const regex: RegExp = /[?]\S/;
+  const indexQueryParam = splitUrl.findIndex((item) => regex.test(item));
+
+  if (indexQueryParam !== -1) {
+    splitUrl.splice(indexQueryParam, 1);
+  }
+
```
Comment (Marc-Roig) [inline packages/plugins/i18n/server/src/register.ts:40]: looks like this breaks in some scenarios:
`isUrlForCreation('/content-manager/collection-types/api::category.category?locale=en')`


what if we did something like this:
```ts
const isUrlForCreation = (url: string) => {
  if (!url) return false;

  // Remove any query params
  // /content-manager/collection-types/api::category.category/?locale=en
  const path = url.split('?')[0];

  // Split path and remove empty strings
  // [ 'content-manager', 'collection-types', 'api::category.category' ]
  const splitUrl = path.split('/').filter(Boolean);

  // Get the last element of the array
  // api::category.category / 1 / publish

  const model = splitUrl[splitUrl.length - 1];

  // If the model contains :: it means it's a uid
  return model.includes('::');
};
```

Comment (Marc-Roig) [reply packages/plugins/i18n/server/src/register.ts:40]: I saw this message resolved but not any commit with a change, do you have any uncommited work?

I would like to include the fix in this week release!

Comment (binar1) [reply packages/plugins/i18n/server/src/register.ts]: Thanks for letting me know. I've just pushed the change.

**Review by Marc-Roig — APPROVED** [review]

**Comment by Marc-Roig** [comment]
This work will be finally released in next week `4.22.0` release

**Comment by nevosol** [comment]
Great job on fixing the issue, thank you!

**Pull request merged**